src/nvim/api/keysets_defs.h

```diff
@@ -172,17 +172,17 @@ typedef struct {
   Boolean altfont;
   Boolean nocombine;
   Boolean default_ DictKey(default);
-  Object cterm;
-  Object foreground;
-  Object fg;
-  Object background;
-  Object bg;
-  Object ctermfg;
-  Object ctermbg;
-  Object special;
-  Object sp;
-  Object link;
-  Object global_link;
+  Union(Integer, String) cterm;
+  Union(Integer, String) foreground;
+  Union(Integer, String) fg;
+  Union(Integer, String) background;
+  Union(Integer, String) bg;
+  Union(Integer, String) ctermfg;
+  Union(Integer, String) ctermbg;
+  Union(Integer, String) special;
+  Union(Integer, String) sp;
+  Union(Integer, String) link;
+  Union(Integer, String) global_link;
```

Comment:
What about the string case?

Reply:
That is being handled. String hl name is converted to int by `api_keydict_to_dict` and friends (otherwise the `HLGroupID` type would have served no purpose)